app/javascript/packs/document-capture-welcome.ts

```diff
@@ -0,0 +1,78 @@
+import { trackEvent } from '@18f/identity-analytics';
+import { hasCamera, isCameraCapableMobile } from '@18f/identity-device';
+
+const GRACE_TIME_FOR_CAMERA_CHECK_MS = 2000;
+const DEVICE_CHECK_EVENT = 'IdV: Mobile device and camera check';
+
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function measure<T>(
+  measureName: string,
+  promise: Promise<T>,
+): Promise<{ result: T; duration: number }> {
+  performance.measure(measureName);
+  const result = await promise;
+  const { duration } = performance.measure(measureName, {
+    end: performance.now(),
+  });
```

Comment:
I'm not super familiar with `performance.measure` and trying to understand its usage. Is it expected that we call it twice in this way? All of the examples I see seem to use a single call.

Alternatively could we create the `start` and `end` values consistently as variables and either subtract, or call `performance.measure`?

```suggestion
  const start = performance.now();
  const result = await promise;
  const end = performance.now();
  const duration = end - start;
```

```suggestion
  const start = performance.now();
  const result = await promise;
  const end = performance.now();
  const { duration } = performance.measure(measureName, { start, end });
```

Reply:
I'd also wonder if it could have any impact to be creating the promise before the `start` time is assigned. I might expect `promise` to passed as a function promise creator.

e.g.

```ts
const start = performance.now();
const result = await createPromise();
const end = performance.now();
```

Reply:
Agreed, I've adjusted to match your recommendations. `performance.measure()` is a real head-scratcher of an API.

Reply:
Agreed, I've adjusted to match your recommendations. `performance.measure()` is a real head-scratcher of an API.